test/e2e/mxGraph.model.test.ts

```diff
@@ -325,63 +350,94 @@ describe('mxGraph model', () => {
-      isUnderline: false,
-      name: 'Arial',
-      size: 11.0,
+    expectModelContainsBpmnEvent('terminateEndEvent', {
+      kind: ShapeBpmnElementKind.EVENT_END,
+      eventKind: ShapeBpmnEventKind.TERMINATE,
+      font: {
+        isBold: false,
+        isItalic: true,
+        isStrikeThrough: false,
+        isUnderline: false,
+        name: 'Arial',
+        size: 11.0,
+      },
+      label: 'Terminate End Event',
     });
-    expectModelContainsBpmnEvent('messageEndEvent', ShapeBpmnElementKind.EVENT_END, ShapeBpmnEventKind.MESSAGE);
+    expectModelContainsBpmnEvent('messageEndEvent', { kind: ShapeBpmnElementKind.EVENT_END, eventKind: ShapeBpmnEventKind.MESSAGE, label: 'Message End Event' });
 
     // throw intermediate event
-    expectModelContainsBpmnEvent('noneIntermediateThrowEvent', ShapeBpmnElementKind.EVENT_INTERMEDIATE_THROW, ShapeBpmnEventKind.NONE, {
-      isBold: false,
-      isItalic: false,
-      isStrikeThrough: true,
-      isUnderline: false,
-      name: 'Arial',
-      size: 11.0,
+    expectModelContainsBpmnEvent('noneIntermediateThrowEvent', {
+      kind: ShapeBpmnElementKind.EVENT_INTERMEDIATE_THROW,
+      eventKind: ShapeBpmnEventKind.NONE,
+      font: {
+        isBold: false,
+        isItalic: false,
+        isStrikeThrough: true,
+        isUnderline: false,
+        name: 'Arial',
+        size: 11.0,
+      },
+      label: 'Throw None Intermediate Event',
+    });
+    expectModelContainsBpmnEvent('messageIntermediateThrowEvent', {
+      kind: ShapeBpmnElementKind.EVENT_INTERMEDIATE_THROW,
+      eventKind: ShapeBpmnEventKind.MESSAGE,
+      label: 'Throw Message Intermediate Event',
     });
-    expectModelContainsBpmnEvent('messageIntermediateThrowEvent', ShapeBpmnElementKind.EVENT_INTERMEDIATE_THROW, ShapeBpmnEventKind.MESSAGE);
 
     // catch intermediate event
-    expectModelContainsBpmnEvent('messageIntermediateCatchEvent', ShapeBpmnElementKind.EVENT_INTERMEDIATE_CATCH, ShapeBpmnEventKind.MESSAGE);
-    expectModelContainsBpmnEvent('IntermediateCatchEvent_Timer_01', ShapeBpmnElementKind.EVENT_INTERMEDIATE_CATCH, ShapeBpmnEventKind.TIMER);
+    expectModelContainsBpmnEvent('messageIntermediateCatchEvent', {
+      kind: ShapeBpmnElementKind.EVENT_INTERMEDIATE_CATCH,
+      eventKind: ShapeBpmnEventKind.MESSAGE,
+      label: 'Catch Message Intermediate Event',
+    });
+    expectModelContainsBpmnEvent('IntermediateCatchEvent_Timer_01', {
+      kind: ShapeBpmnElementKind.EVENT_INTERMEDIATE_CATCH,
+      eventKind: ShapeBpmnEventKind.TIMER,
+      label: 'Timer Intermediate Catch Event',
+    });
 
     // boundary event
-    expectModelContainsBpmnBoundaryEvent('boundary_event_interrupting_message_id', ShapeBpmnEventKind.MESSAGE, true);
-    expectModelContainsBpmnBoundaryEvent('boundary_event_interrupting_timer_id', ShapeBpmnEventKind.TIMER, true);
+    expectModelContainsBpmnBoundaryEvent('boundary_event_interrupting_message_id', {
+      kind: null,
```

Comment:
@csouchet @aibcmars I didn't find a way to avoid passing this `null` value (mandatory in `ExpectedShapeModelElement`). Actually, we could pass any value, as the check always  ensures that we have Boundary event regardless the pass value.
Perhaps it is possible to use Union type to trick this issue: `kind` outside of the `ExpectedShapeModelElement` + mixin. I didn't have time to find the right TypeScript parttern for this use case.

If you have a better solution, feel free to propose a new commit to improve the current implementation

Reply:
I don't know how to this more properly.
Maybe we can fix it later 😉